modmesh/app/euler1d.py

```diff
@@ -132,13 +132,40 @@ class SolverConfig():
           table.
         - :meth:`columnCount()`: Get the number of columns in the
           configuration table.
-        - :meth:`get_var(key_row, key_col)`: Get the value of a configuration
-          variable based on its key.
     """
-    def __init__(self, input_data):
+    class Accessor:
+        """
+        Helper calss to access data within the configuration table using
+        multiple dimensions, currenlty only support 2-dimensions table.
+
+        This class allows users to access data by chaining multiple [],
+        supporting string-based.
+        """
+        def __init__(self, data, dimIdx=0, header=None):
+            self._data = data
+            self._header = header
+            self._dimIdx = dimIdx
+
+        def __getitem__(self, key):
+            if self._dimIdx == 0:
+                for row_idx, row in enumerate(self._data):
+                    if key == row[0]:
+                        return GUIConfig.Accessor(self._data[row_idx],
+                                                  self._dimIdx+1,
+                                                  self._header)
+            else:
+                for idx, ele in enumerate(self._header):
+                    if key == ele:
+                        return self._data[idx]
+            raise KeyError(f'Invaild key: {key} not found in table')
+
+    def __init__(self, input_data, col_headers):
         self.state = State(input_data)
         self._tbl_content = self.state
-        self._col_header = ["variable", "value", "description"]
+        self._col_header = col_headers
+
+    def __getitem__(self, key):
+        return self.Accessor(self._tbl_content, 0, self._col_header)[key]
```

Comment:
Seems that the logic is simple enough to be a function. I am not sure if using a function can reduce the overhead of creating a class?

Reply:
I have the same question.  But I care more about the maintainability.  Both Python functions and classes are slow.  And I don't think we plan to profile for it any time soon.